pysatCDAAC/instruments/cosmic_gps.py

```diff
@@ -104,15 +107,88 @@ def init(self):
+
+    """
+    if self.tag == 'ionprf':
+        # Ionosphere density profiles
+        if self.clean_level == 'clean':
+            # Try and make sure all data is good. Filter out profiles
+            # where source provider processing doesn't do so.
+            # Then get the max density and altitude of this max.
+            self.data = self.data[((self['edmaxalt'] != -999.)
+                                   & (self['edmax'] != -999.))]
+
+            # Make sure edmaxalt is in a "reasonable" range
+            self.data = self.data[((self['edmaxalt'] >= 175.)
+                                   & (self['edmaxalt'] <= 475.))]
+
+            # Remove negative densities
+            for i, profile in enumerate(self['profiles']):
+                # Take out all densities below the highest altitude negative
+                # dens below 325
+                idx, = np.where((profile.ELEC_dens < 0)
+                                & (profile.index <= 325))
+                if len(idx) > 0:
+                    profile.iloc[0:(idx[-1] + 1)] = np.nan
+                # Take out all densities above the lowest altitude negative
+                # dens above 325
+                idx, = np.where((profile.ELEC_dens < 0)
+                                & (profile.index > 325))
+                if len(idx) > 0:
+                    profile.iloc[idx[0]:] = np.nan
+
+                # Do an altitude density gradient check to reduce number of
+                # cycle slips
+                densDiff = profile.ELEC_dens.diff()
+                altDiff = profile.MSL_alt.diff()
+                normGrad = (densDiff / (altDiff * profile.ELEC_dens)).abs()
+                idx, = np.where((normGrad > 1.) & normGrad.notnull())
+                if len(idx) > 0:
+                    self[i, 'edmaxalt'] = np.nan
+                    self[i, 'edmax'] = np.nan
+                    self[i, 'edmaxlat'] = np.nan
+                    profile['ELEC_dens'] *= np.nan
+
+        # Filter out any measurements where things have been set to NaN
+        self.data = self.data[self['edmaxalt'].notnull()]
+
+    elif self.tag == 'scnlv1':
+        # scintillation files
+        if self.clean_level == 'clean':
+            # Make sure all data is good by filtering out profiles where
+            # the source provider processing doesn't work
+            self.data = self.data[((self['alttp_s4max'] != -999.)
+                                   & (self['s4max9sec'] != -999.))]
+
+    return
+
+
+# ----------------------------------------------------------------------------
+# Instrument functions
+#
+
+# Set the list_files routine
+def list_files(tag=None, inst_id=None, data_path=None, format_str=None):
     """Return a Pandas Series of every file for chosen satellite data.
 
     Parameters
     ----------
     tag : string or NoneType
         Denotes type of file to load.
         (default=None)
-    sat_id : string or NoneType
+    inst_id : string or NoneType
```

Comment:
```suggestion
    inst_id : str or NoneType
```

If you use `str` instead of `string`, the eventual docs will create a link to the python docs description for the data type.